### PR TITLE
#414#413 filter model

### LINKED ISSUE #412: Update js service to use query string for the device lists call
references #411

### LINKED ISSUE #413: Add form/dropdowns to allow selection of filtering options
references #411 

Update devices page to have a form that allows users to select filtering parameters.

### LINKED ISSUE #414: Add current filter indicator to show selected filtering options
references #411

### PR DESCRIPTION
**Pull request by Kenny-Miller**
Closes #414
Closes #413 
Closes #412

Adds two dropdowns on the devices page in order to allow filtering of devices based on device type and building.

image

### CODE REVIEW

**Comment by josephmoore99** [comment]
The filter items button feels a little out of place where it's at, I think it might look better if the two filter buttons were closer together?

**Comment by Kenny-Miller** [comment]
> The filter items button feels a little out of place where it's at, I think it might look better if the two filter buttons were closer together?

Something like this? They will still expand horizontally when more selections are made.
image
image

File: BEIMA.Client/src/pages/Devices/DevicesPage.js [hunk 1]
```diff
@@ -1,43 +1,169 @@
...
+              styles={customStyles}
+              ref={buildingSelect}
+            />
+          </div>
+          <Button id="clearFilterButton" onClick={() => {clearFilters(); filterCallback([],[])}}>Clear Filters</Button>
+        </div>
+        <Button id="submitFilterButton" 
+          disabled={loading} 
+          onClick={() => filterCallback(mapSelectedFilters(deviceTypeSelect), mapSelectedFilters(buildingSelect))}
+        >Filter Items</Button>
```
Comment (josephmoore99) [inline BEIMA.Client/src/pages/Devices/DevicesPage.js:124]: I think it makes it more clear if we use "Apply Filters" here

**Comment by josephmoore99** [comment]
> > The filter items button feels a little out of place where it's at, I think it might look better if the two filter buttons were closer together?
> 
> Something like this? They will still expand horizontally when more selections are made. image image

I was thinking more along the lines of having the two buttons next to each other at the end. But idk, UI is hard

**Comment by Kenny-Miller** [comment]
> > > The filter items button feels a little out of place where it's at, I think it might look better if the two filter buttons were closer together?
> > 
> > 
> > Something like this? They will still expand horizontally when more selections are made. image image
> 
> I was thinking more along the lines of having the two buttons next to each other on the same row on the right. But idk, UI is hard, this does look a little better tho

Misread your initial comment. Something like this then? 
image

**Comment by thessbsu** [comment]
That's good, but I would agree to swap the buttons

**Comment by josephmoore99** [comment]
> That's good, but I would agree to swap the buttons

Agreed, also I would use "Apply Filters" instead of "Filter Items".

**Review by Kenny-Miller — COMMENTED** [review]
Finally set of changes:
image

**Review by josephmoore99 — APPROVED** [review]

**Review by thessbsu — APPROVED** [review]

**Review by AshlynAdamson — APPROVED** [review]

**Pull request merged**